hyperion/views/comment_views.py

```diff
@@ -0,0 +1,64 @@
+# pylint: disable=arguments-differ
+
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from rest_framework.response import Response
+from rest_framework import viewsets
+
+from hyperion.authentication import HyperionBasicAuthentication
+from hyperion.serializers import CommentSerializer, UserProfileSerializer
+from hyperion.models import Comment
+from hyperion.models import Post
+
+class CommentViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows users to be viewed or edited.
+    """
+
+    queryset = Comment.objects.all()
+    serializer_class = CommentSerializer
+    authentication_classes = (HyperionBasicAuthentication,)
+    permission_classes = (IsAuthenticated, AllowAny)
+
+    @action(detail=True, methods=["POST"], name="new_comment")
+    def new_comment(self, request, pk=None): # pylint: disable=invalid-name
+        """
+        POST /posts/{post_id}/comments
+        """
+        body = request.data
+        comment_query = body.get("query", None)
+        comment_data = body.get("comment", None)
+        author_jso = comment_data.get("author")
+        author_obj = UserProfileSerializer(data=author_jso)
```

Comment:
You should use `request.user` to retrieve user identify.

Payload to create a new comment shouldn't need an "author" entity.

I've revised the api doc at Notion.